vmm/src/vstate.rs

```diff
@@ -280,6 +292,84 @@ pub struct Vcpu {
 }
 
 impl Vcpu {
+    thread_local!(static TLS_VCPU_PTR: VcpuCell = Cell::new(None));
+
+    /// Associates `self` with the current thread.
+    ///
+    /// Is a prerequisite to succesfully run `run_on_thread_local_vcpu()` on the current thread.
+    /// Will fail if there is already a `Vcpu` present in this thread's TLS.
+    fn init_thread_local_data(&mut self) -> Result<()> {
+        Self::TLS_VCPU_PTR.with(|cell: &VcpuCell| {
+            if cell.get().is_some() {
+                return Err(Error::VcpuTlsInit);
+            }
+            cell.set(Some(self as *mut Vcpu));
+            Ok(())
+        })
+    }
+
+    /// Deassociates `self` from the current thread.
+    ///
+    /// Should be called if the current `self` had called `init_thread_local_data()` and
+    /// now needs to move to a different thread.
+    ///
+    /// Fails if `self` was not previously associated with the current thread.
+    fn reset_thread_local_data(&mut self) -> Result<()> {
```

Comment:
Looking at how `init_thread_local_data` works, what this function does, and how we use it, I keep wondering is there a situation we expect to encounter where it's actually useful? I'd be great if keep code additions to a minimum (and maybe even remove as much old code as necessary as we're refactoring stuff).